Sources/NIOCore/NIOTimer.swift

```diff
@@ -0,0 +1,94 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2024 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+/// A type that handles timer callbacks scheduled with ``EventLoop/setTimer(for:_:)-5e37g``.
+///
+/// - Seealso: ``EventLoop/setTimer(for:_:)-5e37g``.
+public protocol NIOTimerHandler {
+    func timerFired(loop: any EventLoop)
```

Comment:
Another question here is whether we should make this protocol witness generic instead of handling an existential.

Reply:
I can certainly take a look at this. Do we have any examples of API we provide that's generic over an event loop? Everything I've seen so far (`Channel`, `NIOLoopBound`) hold an existential IIUC.

Reply:
Over an event loop is a good question. Probably we should avoid it here, but only because we'll inevitably want to store this in an existential anyway, so we just invite a translation. So let's leave it here.